lib/fog/google/shared.rb

```diff
@@ -97,22 +102,24 @@ def initialize_google_client(options)
             raise ArgumentError.new("Missing required arguments: google_client_email")
           end
 
-          if ENV["DEBUG"]
-            ::Google::Apis.logger = ::Logger.new(::STDERR)
-            ::Google::Apis.logger.level = ::Logger::DEBUG
-          end
-
           auth = ::Google::Auth::ServiceAccountCredentials.make_creds(
             :json_key_io => StringIO.new(json_key_hash.to_json),
             :scope => options[:google_api_scope_url]
           )
         elsif options[:google_auth]
           auth = options[:google_auth]
         else
-          raise ArgumentError.new(
-            "Missing required arguments: google_json_key_location, "\
-            "google_json_key_string or google_auth"
-          )
+          # fall back to Application Default Credentials before erroring
+          begin
+            auth = ::Google::Auth::DefaultCredentials.from_well_known_path(
```

Comment:
When I tested this without any creds on my local workstation, what actually happened is `.from_well_known_path` returned `nil`:
```
[2] pry(#<Fog::Compute::Google::Real>)> auth = ::Google::Auth::DefaultCredentials.from_well_known_path(
[2] pry(#<Fog::Compute::Google::Real>)*   options[:google_api_scope_url]
[2] pry(#<Fog::Compute::Google::Real>)* )
=> nil
```

This caused the auth to fall through and instantiate a broken client:

```
λ bundle exec rake console
[1] pry(main)> connection = Fog::Compute::Google.new(:google_project => "my-project")
=> #<Fog::Compute::Google::Real:70259508535620 @project="my-project" @api_version="v1" @api_url="https://www.googleapis.com/compute/v1/projects/" @compute=#<Google::Apis::ComputeV1::ComputeService:0x00007fcd258a0fd0 @root_url="https://www.googleapis.com/", @base_path="compute/v1/projects/", @upload_path="upload/compute/v1/projects/", @batch_path="batch/compute/v1", @client_options=#<struct Google::Apis::ClientOptions application_name="fog", application_version="1.8.2", proxy_url=nil, open_timeout_sec=nil, read_timeout_sec=nil, send_timeout_sec=nil, log_http_requests=false, transparent_gzip_decompression=true>, @request_options=#<struct Google::Apis::RequestOptions authorization=nil, retries=0, header=nil, normalize_unicode=false, skip_serialization=false, skip_deserialization=false, api_format_version=nil>> @extra_global_projects=[]>
[2] pry(main)> connection.servers
=> #<Fog::Compute::Google::Servers:0x3fe693ee776c>
``` 

I'm not sure which option is the correct one to use but if you're not sure as well - let me know and I'll try to help with this.

Reply:
I looked at the lib again and found the recommended way of loading these creds.